packages/gatsby-core-utils/src/fetch-remote-file.ts

```diff
@@ -72,9 +74,63 @@ const ERROR_CODES_TO_RETRY = [
   `ERR_GOT_REQUEST_ERROR`,
 ]
 
+/********************
+ * Queue Management *
+ ********************/
+
+const GATSBY_CONCURRENT_DOWNLOAD = process.env.GATSBY_CONCURRENT_DOWNLOAD
+  ? parseInt(process.env.GATSBY_CONCURRENT_DOWNLOAD, 10) || 0
+  : 200
+
+const q: queue<IFetchRemoteFileOptions> = Queue(
+  pushToQueue,
+  GATSBY_CONCURRENT_DOWNLOAD
+)
+
+/**
+ * pushToQueue
+ * --
+ * Handle tasks that are pushed in to the Queue
+ */
+async function pushToQueue(
+  task: IFetchRemoteFileOptions,
+  cb: done
+): Promise<unknown> {
+  try {
+    const node = await fetchFile(task)
+    return cb(null, node)
+  } catch (e) {
+    return cb(e)
+  }
+}
+
+/**
+ * pushTask
+ * --
+ * pushes a task in to the Queue and the processing cache
+ *
+ * Promisfy a task in queue
+ * @param {CreateRemoteFileNodePayload} task
+ * @return {Promise<Object>}
+ */
+async function pushTask(task: IFetchRemoteFileOptions): Promise<string> {
+  return new Promise((resolve, reject) => {
+    q.push(task, (err, node) => {
+      if (!err) {
+        resolve(node)
+      } else {
+        reject(err)
+      }
+    })
```

Comment:
Any reason why the `callback API` of `fastq` is used here? I think the async variation is less clutter + we use it with async/promises anyways?

Reply:
It's quite a bit slower in our testing so we avoid it.